src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Bugzilla/Bugzilla52419.cs

```diff
@@ -1,37 +1,47 @@
-﻿#if ANDROID
-using NUnit.Framework;
+﻿using NUnit.Framework;
 using UITest.Appium;
 using UITest.Core;
 
 namespace Microsoft.Maui.TestCases.Tests.Issues;
 
 public class Bugzilla52419 : _IssuesUITest
 {
+#if ANDROID
+	const string TabPage1 = "TAB PAGE 1";
+	const string TabPage2 = "TAB PAGE 2";
+	const string BackButtonIdentifier = "";
+#else
+	const string TabPage1 = "Tab Page 1";
+	const string TabPage2 = "Tab Page 2";
+	const string BackButtonIdentifier = "Tab Page 1";
+#endif
+	const string PushNewPage = "PushNewPage";
+	const string AppearanceLabel = "AppearanceLabel";
+
 	public Bugzilla52419(TestDevice testDevice) : base(testDevice)
 	{
 	}
 
 	public override string Issue => "[A] OnAppearing called for previous pages in a tab's navigation when switching active tabs";
 
-	// [Test]
-	// [Category(UITestCategories.TabbedPage)]
-	// [FailsOnAndroidWhenRunningOnXamarinUITest]
-	// public void Bugzilla52419Test()
-	// {
-	// 	App.WaitForElement(q => q.Marked("Push new page"));
-	// 	App.Tap(q => q.Marked("Push new page"));
-	// 	App.WaitForElement(q => q.Marked("Push new page"));
-	// 	App.Tap(q => q.Marked("Push new page"));
-	// 	App.WaitForElement(q => q.Marked("Push new page"));
-	// 	App.Tap(q => q.Marked("Push new page"));
-	// 	App.Tap(q => q.Marked("Tab Page 2"));
-	// 	App.Tap(q => q.Marked("Tab Page 1"));
-	// 	App.Tap(q => q.Marked("Tab Page 2"));
-	// 	App.Tap(q => q.Marked("Tab Page 1"));
-	// 	App.Back();
-	// 	App.WaitForElement(q => q.Marked("AppearanceLabel"));
-	// 	var label = App.Query(q => q.Marked("AppearanceLabel"))[0];
-	// 	Assert.AreEqual("Times Appeared: 2", label.Text);
-	// }
+	 [Test]
+	 [Category(UITestCategories.TabbedPage)]
+	 public void Bugzilla52419Test()
+	 {
+		App.WaitForElementTillPageNavigationSettled(PushNewPage);
+	 	App.Tap(PushNewPage);
+		App.WaitForElementTillPageNavigationSettled(PushNewPage);
+	 	App.Tap(PushNewPage);
+		App.WaitForElementTillPageNavigationSettled(PushNewPage);
+	 	App.Tap(PushNewPage);
+		App.WaitForElementTillPageNavigationSettled(TabPage2);
+	 	App.Tap(TabPage2);
```

Comment:
Mnnn, maybe I could include a **TapTab** method that internally does the wait, and also in the case of Android the _ToUpperCase_.